src/main/scala/ztis/cassandra/CassandraClient.scala

```diff
@@ -12,39 +12,70 @@ class CassandraClient(private[cassandra] val config: CassandraConfiguration) ext
     .build()
   private val session = cluster.connect()
 
-  session.execute(CassandraClient.createKeyspaceQuery(config.keyspace))
-  session.execute(CassandraClient.createRatingsTableQuery(config.keyspace, config.ratingsTableName))
-  session.execute(CassandraClient.createRelationshipsTableQuery(config.keyspace, config.relationshipsTableName))
+  initializeDatabase()
   val preparedInsertToRatings = session.prepare(CassandraClient.insertToRatingsQuery(config.keyspace, config.ratingsTableName))
   val preparedInsertToRelationships = session.prepare(CassandraClient.insertToRelationshipsQuery(config.keyspace, config.relationshipsTableName))
-  
-  def updateRating(userVideoRating: UserVideoRating): Unit = {
+
+  /**
+   * Adds new rating for user, also adds new association to all users following given user.
+   * 
+   * This method assumes that all following users have the same userOrigin as user.
+   * @param userVideoRating
+   * @param usersFollowing
+   */
+  def updateRating(userVideoRating: UserVideoRating, usersFollowing: Vector[Int] = Vector.empty[Int]): Unit = {
     val userID: java.lang.Integer = userVideoRating.userID
+    val userOrigin = userVideoRating.userOrigin.toString
     val videoID: java.lang.Integer = userVideoRating.videoID
+    val videoOrigin = userVideoRating.videoOrigin.toString
     val rating: java.lang.Integer = userVideoRating.rating
 
     val statement = preparedInsertToRatings.bind(userID,
-      userVideoRating.userOrigin.toString,
+      userOrigin,
       videoID,
-      userVideoRating.videoOrigin.toString,
+      videoOrigin,
       rating)
     session.execute(statement)
+    
+    usersFollowing.foreach { followingUserID =>
+      updateImplicitAssociation(followingUserID, userOrigin, userID, userOrigin, videoID, videoOrigin) 
+    }
   }
 
   def updateImplicitAssociation(userVideoImplicitAssociation: UserVideoImplicitAssociation): Unit = {
     val userID: java.lang.Integer = userVideoImplicitAssociation.internalUserID
     val followedUserID: java.lang.Integer = userVideoImplicitAssociation.followedInternalUserID
     val videoID: java.lang.Integer = userVideoImplicitAssociation.internalVideoID
-    
-    val statement = preparedInsertToRelationships.bind(userID,
+
+    updateImplicitAssociation(userID,
       userVideoImplicitAssociation.userOrigin.toString,
       followedUserID,
       userVideoImplicitAssociation.followedUserOrigin.toString,
       videoID,
       userVideoImplicitAssociation.videoOrigin.toString)
+  }
+  
+  private def updateImplicitAssociation(userID: java.lang.Integer, 
```

Comment:
We are using this private version inside updateRating - we do not need to create unnecessary  UserVideoImplicitAssociation objects